app/services/sso/basic_with_hidden.rb

```diff
@@ -0,0 +1,11 @@
+module SSO
+  class BasicWithHidden < Basic
+    def available?
+      controller.api_request? && http_auth_set? && controller.is_a?(Katello::Api::Rhsm::CandlepinProxiesController)
+    end
+
+    def current_user
+      User.unscoped.find_by_login(self.user)
```

Comment:
This is only for authentication, authorization is done separately and requires proper permissions but I guess we could add check for the hidden auth source.